improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -450,78 +484,99 @@ def _probabilities_to_percentiles(
 
         # Invert probabilities for data thresholded above thresholds.
-        relation = find_threshold_coordinate(
-            forecast_probabilities).attributes['spp__relative_to_threshold']
-        if relation == 'above':
+        relation = find_threshold_coordinate(forecast_probabilities).attributes[
+            "spp__relative_to_threshold"
+        ]
+        if relation == "above":
             probabilities_for_cdf = 1 - prob_slices
-        elif relation == 'below':
+        elif relation == "below":
             probabilities_for_cdf = prob_slices
         else:
-            msg = ("Probabilities to percentiles only implemented for "
-                   "thresholds above or below a given value."
-                   "The relation to threshold is given as {}".format(relation))
+            msg = (
+                "Probabilities to percentiles only implemented for "
+                "thresholds above or below a given value."
+                "The relation to threshold is given as {}".format(relation)
+            )
             raise NotImplementedError(msg)
 
-        threshold_points, probabilities_for_cdf = (
-            self._add_bounds_to_thresholds_and_probabilities(
-                threshold_points, probabilities_for_cdf, bounds_pairing))
+        (
+            threshold_points,
+            probabilities_for_cdf,
+        ) = self._add_bounds_to_thresholds_and_probabilities(
+            threshold_points, probabilities_for_cdf, bounds_pairing
+        )
 
         if np.any(np.diff(probabilities_for_cdf) < 0):
-            msg = ("The probability values used to construct the "
-                   "Cumulative Distribution Function (CDF) "
-                   "must be ascending i.e. in order to yield "
-                   "a monotonically increasing CDF."
-                   "The probabilities are {}".format(probabilities_for_cdf))
+            msg = (
+                "The probability values used to construct the "
+                "Cumulative Distribution Function (CDF) "
+                "must be ascending i.e. in order to yield "
+                "a monotonically increasing CDF."
+                "The probabilities are {}".format(probabilities_for_cdf)
+            )
             warnings.warn(msg)
 
         # Convert percentiles into fractions.
         percentiles_as_fractions = np.array(
-            [x/100.0 for x in percentiles], dtype=np.float32)
-
-        forecast_at_percentiles = (
-            np.empty((len(percentiles),
-                      probabilities_for_cdf.shape[0]), dtype=np.float32)) \
-            # pylint: disable=unsubscriptable-object
-        for index in range(probabilities_for_cdf.shape[0]): \
-                # pylint: disable=unsubscriptable-object
+            [x / 100.0 for x in percentiles], dtype=np.float32
+        )
+
+        forecast_at_percentiles = np.empty(
+            (len(percentiles), probabilities_for_cdf.shape[0]), dtype=np.float32
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Value 'probabilities_for_cdf.shape' is unsubscriptable](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5354756)